website/content/en/highlights/2022-03-22-0-21-0-upgrade-guide.md

```diff
@@ -28,6 +29,32 @@ We cover them below to help you upgrade quickly:
 
 ### Breaking changes
 
+#### Syntax changes for non-VRL paths {#path-syntax-changes}
+
+Previously, there were two different ways to describe paths. VRL (remap transforms) use a newer syntax, while everything
+else in Vector (templating / all transforms except remap) still used an older syntax. This was a constant pain point for users,
+and we have taken some steps to migrate towards the VRL syntax. This is a **breaking change** that may require migration.
+
+The old syntax was very lenient in the characters that were allowed in a field name. It also supported single character escapes.
+The new syntax only allows capital/lowercase letters, numbers, and underscore. Any other character will require the field name to be quoted.
+Quotes around a field name replace single character escaping. This brings the old syntax in line with the newer (VRL) syntax, except for support
+of the "coalescing" operator. The new parser is also faster, and contributed to the increased performance you may see in this release.
+
+Migration will be required for any paths used outside a VRL context. That is any transform (except remap), templating,
+and any source or sink referencing field names. There are _no_ changes to the VRL syntax.
+
+Here are some examples that require migrating
+
+| old syntax | new syntax | comment |
+| ------------- | --------------- | -------------- |
+| foo\\.bar.baz     | "foo.bar".baz     | The `.` field separator needs to be escaped if used as part of a field name. The old syntax allowed individual character escaping. The new syntax requires quotes around the field name.
+| headers.User-Agent | headers."User-Agent" | `-` requires quotes with the new syntax
+| foo with spaces | "foo with spaces" | Spaces also need to be quoted
```

Comment:
It might be worth giving an actual TOML example (e.g. use a common source/transform/sink with the before/after config), as the field quoting means you need to take care to use existing double/single quotes correctly.

Reply:
Yeah, agreed, users do get confused about the interactions between TOML escaping.